flink-table-store-core/src/main/java/org/apache/flink/table/store/file/schema/SchemaManager.java

```diff
@@ -201,6 +203,12 @@ public TableSchema commitChanges(List<SchemaChange> changes) throws Exception {
                                 DataType newType =
                                         TableSchema.toDataType(
                                                 update.newLogicalType(), new AtomicInteger(0));
+                                checkState(
+                                        LogicalTypeCasts.supportsImplicitCast(
+                                                field.type().logicalType, update.newLogicalType()),
+                                        String.format(
+                                                "Row type %s cannot be converted to %s without loosing information.",
```

Comment:
Could this add the field name in the message which is convenient for users to check which field type is incorrect？